api/app/controllers/spree/api/address_books_controller.rb

```diff
@@ -2,32 +2,41 @@ module Spree
   module Api
     class AddressBooksController < Spree::Api::BaseController
       # Note: the AddressBook is the resource to think about here, not individual addresses
+      before_filter :authorize_user_address
+      before_filter :load_user_addresses
 
       def show
-        render_address_book
+        render :show, status: :ok
       end
 
       def update
         address_params = address_book_params
         default_flag = address_params.delete(:default)
-        address = current_api_user.save_in_address_book(address_params, default_flag)
+        address = address_book_user.save_in_address_book(address_params, default_flag)
         if address.valid?
-          render_address_book
+          render :show, status: :ok
         else
           invalid_resource!(address)
         end
       end
 
       def destroy
-        current_api_user.remove_from_address_book(params[:address_id])
-        render_address_book
+        address_book_user.remove_from_address_book(params[:address_id])
+        render :show, status: :ok
       end
 
       private
 
-      def render_address_book
-        @user_addresses = current_api_user.user_addresses
-        render :show, status: :ok
+      def authorize_user_address
+        authorize! :manage, UserAddress.new(user_id: address_book_user.id)
```

Comment:
@gmacdougall this situation is a bit strange because there is no "AddressBook" model to validate against and we perform actions through the address book interface rather operating on the ActiveRecord models directly.  e.g., for deletion we're deleting (actually _archiving_) a `UserAddress` but we never have that UserAddress in hand at this level.  We just have an address book interface through which we request a deletion (aka archival).

@gvaughn talked in the solidus channel earlier about possibly changing things so that we have an actual UserAddressBook model, though that would take some work.

> if address_book_user is nil, this will throw an odd 500 exception

`address_book_user` raises an `ActiveRecord::RecordNotFound` rather than returning nil, and `ActiveRecord::RecordNotFound` errors are converted into 404s aren't they?